server/src/main/java/com/cloud/network/router/NetworkHelperImpl.java

```diff
@@ -618,7 +616,7 @@ protected List<HypervisorType> getHypervisors(final RouterDeploymentDefinition r
             throw new InsufficientServerCapacityException("Unable to create virtual router, there are no clusters in the zone." + getNoHypervisorsErrMsgDetails(),
                     DataCenter.class, dest.getDataCenter().getId());
         }
-        return hypervisors;
+        return new ArrayList(new LinkedHashSet<>(hypervisors));
```

Comment:
it is strange that `hypervisors` is not a `Set` in the first place, but this will work.

Reply:
@DaanHoogland 
right.
I was thinking about it, but if we define it as a Set, we need to convert the following from List to Set
```
            hypervisors = _resourceMgr.getSupportedHypervisorTypes(dest.getDataCenter().getId(), true, routerDeploymentDefinition.getPlan().getPodId());
```

Reply:
yes, I also saw another call being done in the method that might require changes: `filterSupportedHypervisors(final List<HypervisorType> hypervisors)`.

I don't care if we leave that for later.

Reply:
> yes, I also saw another call being done in the method that might require changes: `filterSupportedHypervisors(final List<HypervisorType> hypervisors)`.
> 
> I don't care if we leave that for later.

yep, the method did nothing